lib/influx.js

```diff
@@ -0,0 +1,35 @@
+/*
+* Sends request metrics to InfluxDB
+* https://docs.influxdata.com/enterprise_influxdb/v1/guides/write_data/
+*/
+export const reportMetric = async (request, env) => {
+  // Define API endpoint and headers
+  const url = `${env.INFLUX_URL}/api/v2/write?&bucket=${env.INFLUX_DATABASE}&precision=ms`;
+
+  return fetch(url, {
+    method: 'POST',
+    headers: {
+      'Authorization': `Token ${env.INFLUX_TOKEN}`,
+      'Content-Type': 'application/octet-stream'
+    },
+    body: createMetricsFromRequest(request, env),
+  })
+}
```

Comment:
I find this function more challenging to read because it mixes low-level concerns about writing data points to InfluxDB with high-level details of what steps are needed to report an incoming request. In particular, `body: createMetricsFromRequest(request, env)` comes as a surprise to me, I wouldn't expect the decision about what request body bytes to write at the point where we define fetch options.

Proposed improvement:

```js
export const reportRequestMetric = async (request, env) => {
  await writeMetrics(createMetricsFromRequest(request, env), env)
}

const writeMetric = async (lineProtocolData, env) => {
  // Define API endpoint and headers
  const url = `${env.INFLUX_URL}/api/v2/write?&bucket=${env.INFLUX_DATABASE}&precision=ms`;
return fetch(url, {
    method: 'POST',
    headers: {
      'Authorization': `Token ${env.INFLUX_TOKEN}`,
      'Content-Type': 'application/octet-stream'
    },
    body: lineProtocolData
  })
}
```

Having written that, this file is only 30 lines of code now, it's not that difficult to read all of it to build a good understanding of what's going on, so my suggestion is a "nice to have" improvement.